### PR TITLE
Add driver command to update log config

### LINKED ISSUE #115: Add command to update logging configuration
This was merged and released a litttle while ago: https://github.com/zwave-js/node-zwave-js/pull/1754

It would be great to add support for this in the server as well as in the Python library so we can add a service in Home Assistant to update the log level. This would eliminate one of the common reasons why users currently restart server.

### PR DESCRIPTION
**Pull request by raman325**
Related to https://github.com/zwave-js/zwave-js-server/issues/115 and https://github.com/zwave-js/node-zwave-js/pull/1754. This will allow us to add support to update the log configuration in the Python library and then expose a service in Home Assistant to update the log levels without having to restart the server.

The version bump is a breaking change to the add-on - updating the add-on would depend on https://github.com/home-assistant/core/pull/46355 in order for `zwave_js` thermostats in Home Assistant to continue to function.

Resolves #115

### CODE REVIEW

File: package.json
Comment (MartinHjelmare) [inline package.json:28]: Let's do separate PRs for bumping zwave-js, the log feature addition and bumping the server version.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

**Comment by raman325** [comment]
I will rebase once #118 is merged

**Comment by MartinHjelmare** [comment]
#118 is merged. :+1:

**Comment by MartinHjelmare** [comment]
I'd like some TS/JS eyes on this PR.

**Comment by raman325** [comment]
> I'd like some TS/JS eyes on this PR.

Do we want to bump the server version now, or wait until this is merged?

**Comment by MartinHjelmare** [comment]
We can do a release now of the server and have this PR go in a later release.

File: README.md
Comment (balloob) [inline README.md:142]: Can we expand the `Transport` type, not sure what it is here.

File: src/lib/outgoing_message.ts
Comment (balloob) [inline src/lib/outgoing_message.ts:34]: I think that this should be an `|` ?

```suggestion
} | {
  update_log_config: Record<string, never>;
} | NodeResultTypes |
```

I also think that we want to extract the 2 anonymous defined interfaces into named interfaces.

File: src/lib/incoming_message.ts
Comment (balloob) [inline src/lib/incoming_message.ts:15]: Why is this `DeepPartial` and not `Partial` ? Can partial transport configs be defined ?

Comment (raman325) [reply src/lib/incoming_message.ts:15]: I used the type that is expected upstream: https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/driver/Driver.ts#L590

Will `Partial` work instead?

I don't know anything about the `Transport` part of the config but it seems that is mainly for internal use, e.g. it shouldn't be updated externally (https://github.com/zwave-js/node-zwave-js/blob/master/packages/core/src/log/shared.ts#L129)

File: src/lib/outgoing_message.ts
Comment (raman325) [reply src/lib/outgoing_message.ts:34]: I followed the existing convention by using `&` but `|` makes more sense if I understand this correctly. How was this not broken before?

File: README.md
Comment (raman325) [reply README.md:142]: I'm wondering if we should just drop it altogether (drop it from the `node-zwave-js` response before serialization)? As I mentioned in another PR, I think this is managed entirely internally. From what I can tell, transports are really just a stream writer that is created based on the other parameter values.

Comment (balloob) [reply README.md:142]: yeah, in that case let's not expose more than we need right now for HA.

File: src/lib/outgoing_message.ts
Comment (balloob) [reply src/lib/outgoing_message.ts:34]: I think it's just used for typing but probably not very useful haha

Comment (raman325) [reply src/lib/outgoing_message.ts]: looks like `&` is right, `|` results in type errors

File: src/lib/incoming_message.ts
Comment (raman325) [reply src/lib/incoming_message.ts]: I looked into `DeepPartial` and `Partial` and since we have decided not to support updating transports, I don't think `DeepPartial` is needed

**Comment by raman325** [comment]
Since we are on the latest version of `node-zwave-js`, this PR can be merged with the other one. Closing this one since that one is more complete

**Pull request closed without merging**
